spec/support/matchers/have_providers.rb

```diff
@@ -0,0 +1,39 @@
+RSpec::Matchers.define :have_providers do |*providers|
+  providers = providers.flatten
```

Comment:
You could use `Array#flatten!` to flatten in place here. Same in the other matcher.